app/main/views/suppliers.py

```diff
@@ -86,6 +88,119 @@ def download_agreement_file(supplier_id, framework_slug, document_name):
     return redirect(url)
 
 
+@main.route('/suppliers/<supplier_id>/countersigned-agreements/<framework_slug>/'
+            '<document_name>', methods=['GET'])
+@login_required
+@role_required('admin-ccs-sourcing')
+def list_countersigned_agreement_file(supplier_id, framework_slug, document_name):
```

Comment:
I think this will only ever show either zero or one countersigned agreements. I think that the table is a good way of displaying it but there seems to be logic in here and in download countersigned agreement file that assumes there can be multiple.